ner_v2/detectors/numeral/utils.py

```diff
@@ -0,0 +1,79 @@
+
+def get_number_from_numerals(text, number_word_dict):
```

Comment:
This implementation is very hard to understand and the branches of if-else and resetting multiple variables in- between the loop causes more confusion and can lead to bugs. Also, this needs code duplication at the end after the loop. Overall I feel there is a lot this can be improved. Maybe we'll need to take a decision when to improve this.

Keeping the implementation aside, since this also eliminates extra whitespace from the original text, I have some suggestions to avoid that here: https://gist.github.com/chiragjn/d0e32f81b56d66a7d2b248168e7320b6/revisions. Please go through it and check if it works for our case.

Reply:
From the top of my head, there are 4 types of patterns

`(unit scale+ ignore_tokens?)+ unit?`

- three thousand two hundred and ten = 3210
- three hundred thousand = 300000

There needs to be one constraint. Multiplying consecutive scales and considering them as a single scale, each successive scale must smaller than the one occurs before it (million, thousand, hundred)
-----

`scale+`

- hundred thousand = 1,00,0000

One is assumed in this case
-----

`10s_unit unit`

- Twenty three = 23
Maybe this applies to only some languages not all

-------

`unit `

- one = 1
- two = 2
- fifty = 50

We might need to discuss what works and what doesn't

Reply:
```
[In]   >> number_words_map = {'one': (1, 1), 'thousand':  (1000, 0)}
[In]   >> get_number_from_numerals('one thousand', number_words_map)``
[Out] >> ([1, 1000], [' one', ' thousand'])```

Not working properly